packages/tokens/repositories/dynamo_token_repository.go

```diff
@@ -1,57 +1,34 @@
 package repositories
 
 import (
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/dynamodb"
-	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
 	"github.com/juju/errors"
+	"theodo.red/creditcompanion/packages/database"
+	"theodo.red/creditcompanion/packages/database/marsh"
+	"theodo.red/creditcompanion/packages/database/tdynamo"
 	"theodo.red/creditcompanion/packages/tokens/models"
 )
 
-type DynamoTokenRepository struct {
-	db        DynamoDbInterface
-	tableName string
+type TokenRepository interface {
+	Get(id string) (*models.Token, error)
 }
 
-type DynamoDbInterface interface {
-	GetItem(*dynamodb.GetItemInput) (*dynamodb.GetItemOutput, error)
+type DynamoTokenRepository struct {
+	repo database.Repository
 }
 
 func (r *DynamoTokenRepository) Get(id string) (*models.Token, error) {
-	token := &models.Token{}
-
-	result, getErr := r.db.GetItem(&dynamodb.GetItemInput{
-		TableName: aws.String(r.tableName),
-		Key: map[string]*dynamodb.AttributeValue{
-			"Id": {
-				S: &id,
-			},
-		}})
-	if getErr != nil {
-		// Don't log errors til you handle them
-		return nil, errors.Annotate(getErr, "Call to retrieve item from db failed.")
+	token := new(models.Token)
+	err := r.repo.Get(id, token)
+	if err != nil {
+		return nil, errors.Annotate(err, "Call to marshalling repo failed.")
 	}
-
-	if result.Item == nil {
-		return nil, errors.New("Token " + id + " could not be found.")
-	}
-
-	unmarshalErr := dynamodbattribute.UnmarshalMap(result.Item, &token)
-	if unmarshalErr != nil || (models.Token{}) == *token {
-		return nil, errors.New("Failed to unmarshal token " + id)
-	}
-
 	return token, nil
 }
 
-func (r *DynamoTokenRepository) Set(id string, token *models.Token) error {
-	// TODO: Implement this.
-	return nil
-}
-
-func NewDynamoTokenRepository(db DynamoDbInterface) *DynamoTokenRepository {
+func NewTokenRepository(db tdynamo.DynamoDbInterface) TokenRepository {
```

Comment:
Consequently, change this to NewDynamoTokenRepository given this _is_ tied to dynamo